### ISSUE TITLE
Ignore firmware loading failure if retrial is successful

### ISSUE THREAD

**Issue by fredoh9** [thesofproject/sof-test#340]
FW load may fail at any time, we have up to 3 time retrials. If FW loading file fails, below errors cause test case failure.

```
kernel: [  636.613727] sof-audio-pci 0000:00:1f.3: unknown ROM status value 06000010
kernel: [  636.613734] sof-audio-pci 0000:00:1f.3: error: status = 0x00000000 panic = 0x00000000
kernel: [  636.613768] sof-audio-pci 0000:00:1f.3: error: extended rom status:0x6000010 0x0 0x0 0x0 0x0 0x0 0x2001180 0x0 
kernel: [  636.613773] sof-audio-pci 0000:00:1f.3: unknown ROM status value 06000010
```
We can ignore, but I'm afraid status and panic codes are NOT specific enough.

- "0000:00:1f.3: error: status = 0x00000000 panic = 0x00000000"
- "0000:00:1f.3: error: extended rom status:0x6000010"

Logically we should check iteration, `Attempting iteration 1 of Core En/ROM load...` and firmware boot fine `firmware boot complete` then ignore previous above errors. This is some work to do in current error checking.

https://sof-ci.01.org/linuxpr/PR2359/build4303/devicetest/

```
kernel: [  636.311495] sof-audio-pci 0000:00:1f.3: loading firmware
kernel: [  636.311518] sof-audio-pci 0000:00:1f.3: Debug PCIR: 00000002 at  00000048
kernel: [  636.311539] sof-audio-pci 0000:00:1f.3: Debug PCIW: 00000000 at  00000048
kernel: [  636.311545] sof-audio-pci 0000:00:1f.3: Debug PCIR: 00000000 at  00000044
kernel: [  636.311551] sof-audio-pci 0000:00:1f.3: Debug PCIW: 00000004 at  00000044
kernel: [  636.311552] sof-audio-pci 0000:00:1f.3: booting DSP firmware
kernel: [  636.311638] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x40000 successful
kernel: [  636.311689] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x40000 successful
kernel: [  636.311695] sof-audio-pci 0000:00:1f.3: period_bytes:0x0
kernel: [  636.311697] sof-audio-pci 0000:00:1f.3: periods:1
kernel: [  636.311794] sof-audio-pci 0000:00:1f.3: Attempting iteration 0 of Core En/ROM load...
kernel: [  636.312361] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf0f0f0f successful
kernel: [  636.312377] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf0f0f0e successful
kernel: [  636.312380] sof-audio-pci 0000:00:1f.3: unstall/run core: core_mask = 1
kernel: [  636.312386] sof-audio-pci 0000:00:1f.3: DSP core(s) enabled? 1 : core_mask 1
kernel: [  636.313249] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x80000000 successful
kernel: [  636.313272] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf010e0e successful
kernel: [  636.613721] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x6000010 timedout
kernel: [  636.613727] sof-audio-pci 0000:00:1f.3: unknown ROM status value 06000010
kernel: [  636.613734] sof-audio-pci 0000:00:1f.3: error: status = 0x00000000 panic = 0x00000000
kernel: [  636.613768] sof-audio-pci 0000:00:1f.3: error: extended rom status:0x6000010 0x0 0x0 0x0 0x0 0x0 0x2001180 0x0 
kernel: [  636.613773] sof-audio-pci 0000:00:1f.3: unknown ROM status value 06000010
kernel: [  636.613779] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x1010f0f successful
kernel: [  636.613784] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x1000f0f successful
kernel: [  636.613786] sof-audio-pci 0000:00:1f.3: DSP core(s) enabled? 0 : core_mask f
kernel: [  636.613788] sof-audio-pci 0000:00:1f.3: Attempting iteration 1 of Core En/ROM load...
kernel: [  636.613793] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf0f0f0f successful
kernel: [  636.613807] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf0f0f0e successful
kernel: [  636.613809] sof-audio-pci 0000:00:1f.3: unstall/run core: core_mask = 1
kernel: [  636.613813] sof-audio-pci 0000:00:1f.3: DSP core(s) enabled? 1 : core_mask 1
kernel: [  636.614369] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x80000000 successful
kernel: [  636.614391] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf010e0e successful
kernel: [  636.689725] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x5000001 successful
kernel: [  636.703244] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x5 successful
kernel: [  636.703726] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x140000 successful
kernel: [  636.703731] sof-audio-pci 0000:00:1f.3: Firmware download successful, booting...
kernel: [  636.713934] sof-audio-pci 0000:00:1f.3: ipc rx: 0x70000000: FW_READY
kernel: [  636.713938] sof-audio-pci 0000:00:1f.3: ipc: DSP is ready 0x70000000 offset 0x81000
kernel: [  636.713944] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x70000000: FW_READY
kernel: [  636.714052] sof-audio-pci 0000:00:1f.3: firmware boot complete
```

**Comment by fredoh9** [comment]
Found same problem in check-runtime-pm-status.sh with ICL_RVP_HDA at [sof-daily-report] 2020-08-17 linux/for-next + master by SOF-CI.

**Comment by fredoh9** [comment]
Pierre submit a PR to reduce verbosity of boot error logs. He said a couple remaining.
https://github.com/thesofproject/linux/pull/2382

**Comment by xiulipan** [comment]
Have an update to our filter to our update https://github.com/thesofproject/sof-test/pull/355

**Comment by marc-hb** [comment]
@fredoh9  did you see this outside ICL? If not maybe #3395 is a duplicate and we can close this one (#3395 has more information)

**Comment by aiChaoSONG** [comment]
Close for old filter is added back to our sof-test ignore list.
We have patch to fix this issue on linux-5.9.
